src/rpc/util.h

```diff
@@ -157,6 +157,30 @@ struct RPCArg {
     const std::string m_oneline_description; //!< Should be empty unless it is supposed to override the auto-generated summary line
     const std::vector<std::string> m_type_str; //!< Should be empty unless it is supposed to override the auto-generated type strings. Vector length is either 0 or 2, m_type_str.at(0) will override the type of the value in a key-value pair, m_type_str.at(1) will override the type in the argument description.
 
+    void CheckFallback() {
+        if (m_fallback.index() != 2) return;
+        switch (std::get<Default>(m_fallback).getType()) {
+        case UniValue::VOBJ:
+            CHECK_NONFATAL(m_type == Type::OBJ);
+            break;
+        case UniValue::VARR:
+            CHECK_NONFATAL(m_type == Type::ARR);
+            break;
+        case UniValue::VSTR:
+            CHECK_NONFATAL(m_type == Type::STR || m_type == Type::STR_HEX || m_type == Type::AMOUNT);
+            break;
+        case UniValue::VNUM:
+            CHECK_NONFATAL(m_type == Type::NUM || m_type == Type::AMOUNT || m_type == Type::RANGE);
+            break;
+        case UniValue::VBOOL:
+            CHECK_NONFATAL(m_type == Type::BOOL);
+            break;
+        case UniValue::VNULL:
+        default:
+            break;
+        }
+    }
```

Comment:
I am not really a fan of putting a 20-line function in a header

Reply:
Yeah I made the smallest change to test this approach. I’ll move this.